qiskit/circuit/library/arithmetic/inverse_chebyshev.py

```diff
@@ -0,0 +1,194 @@
+        """Construct the piecewise Chebyshev approximation of the inverse function.
+
+        Args:
+            num_state_qubits: number of qubits representing the state.
+            epsilon: accuracy of the approximation.
+                Defaults to ``1e-2``.
+            constant: :math:`C` in :math:`arcsin(C/x)`.
+                Defaults to ``1``.
+            kappa: condition number of the system.
+                Defaults to ``1``.
+            name: The name of the circuit object.
+        """
+        super().__init__(name=name)
+
+        # define internal parameters
+        self._num_state_qubits = None
+
+        # Store parameters
+        self._epsilon = epsilon if epsilon is not None else 1e-2
+        self._kappa = kappa if kappa is not None else 1
+        self._constant = constant if constant is not None else 1
+
+        # We perform the identity operation on [1,a].
+        # int(round()) necessary to compensate for computer precision.
+        self._N_l = 2 ** num_state_qubits
+        self._a = int(round(self._N_l ** (2/3)))
+
+        # Calculate the degree of the polynomial and the number of intervals
+        r = 2 * self._constant / self._a + np.sqrt(np.abs(1 - (2 * self._constant / self._a) ** 2))
+        self._degree = int(np.log(1 + (16.23 * np.sqrt(np.log(r) ** 2 + (np.pi/2) ** 2) *
+                                       self._kappa * (2 * self._kappa - self._epsilon)) /
+                                  self._epsilon))
+        self._num_intervals = int(np.ceil(np.log((self._N_l - 1) / self._a) / np.log(5)))
+
+        # Calculate breakpoints and polynomials
+        self._breakpoints = []
+        self._polynomials = []
+        for i in range(0, self._num_intervals):
+            # Add the breakpoint to the list
+            self._breakpoints.append(self._a * (5 ** i))
+            # Define the right breakpoint of the interval
+            if i == self._num_intervals - 1:
+                r_breakpoint = self._N_l - 1
+            else:
+                r_breakpoint = self._a * (5 ** (i + 1))
+            # Calculate the polynomial approximating the function on the current interval
+            poly = Chebyshev.interpolate(lambda x: np.arcsin(self._constant / x), self._degree,
+                                         domain=[self._breakpoints[i], r_breakpoint])
+            # Convert polynomial to the standard basis and rescale it for the rotation gates
+            poly = 2 * poly.convert(kind=np.polynomial.Polynomial).coef
+            # Convert to list and append
+            self._polynomials.append(poly.tolist())
+
+        self._poly_r = PiecewisePolynomialPauliRotations(num_state_qubits, self._breakpoints,
+                                                         self._polynomials)
+
+        self.num_state_qubits = num_state_qubits
+
+    @property
+    def num_ancilla_qubits(self):
+        """Return the number of required ancillas."""
+        return self._poly_r.num_ancilla_qubits
```

Comment:
This should not be necessary, if you use the `AncillaRegister` to define the ancillas, you can query the number of ancillas via the built-in circuit method `num_ancillas`